editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -2599,6 +2599,9 @@ void CanvasItemEditor::_gui_input_viewport(const Ref<InputEvent> &p_event) {
 
 void CanvasItemEditor::_update_cursor() {
 	CursorShape c = CURSOR_ARROW;
+	bool should_switch = false;
+	if (drag_selection.size() != 0)
+		should_switch = abs(fmod(drag_selection[0]->_edit_get_rotation(), Math_PI)) >= Math_PI / 4 && abs(fmod(drag_selection[0]->_edit_get_rotation(), Math_PI)) <= Math_PI - Math_PI / 4;
```

Comment:
```suggestion
	if (drag_selection.size() != 0) {
		should_switch = abs(fmod(drag_selection[0]->_edit_get_rotation(), Math_PI)) >= Math_PI / 4 && abs(fmod(drag_selection[0]->_edit_get_rotation(), Math_PI)) <= Math_PI - Math_PI / 4;
	}
```
There's probably a simpler way to know if it's flipped using dot product or similar, but I'm too tired to think it through now ;)
CC @groud @aaronfranke

Reply:
Yes, something like `should_switch = abs(x.normalized().dot(Vector2(1, 0))) < Math_SQRT12;` should do it, where `x` is the X vector of the transformation matrix.

Reply:
I feel like using the transformation and the dot product is a lot less readable, and not especially faster. If you insist, I'll change it, but I think I have a clearer way with fmod being on Math_PI / 4 (I'll commit it later, really busy atm)

Reply:
Is calling `_edit_get_rotation` an efficient operation? Also, why `fmod` with `Math_PI / 4` (quarter turn)? I don't see how that would help, since it means that one eighth turn behaves the same as zero turns and the same as one quarter turn.

Reply:
I figured out the correct formula that's a chunk shorter. I'll put it in tomorrow (again, super busy. In the middle of exams)